lib/nimbusRecipe.ts

```diff
@@ -79,33 +79,45 @@ export class NimbusRecipe implements NimbusRecipeType {
       case "aboutwelcome":
 
         branchInfo.id = feature.value.id
-
-        // featureValue will become the "content" object in a spotlight JSON
-        let spotlightFake = {
-          id: this._rawRecipe.id,
-          template: "spotlight",
-          targeting: true,
-          content: feature.value,
-        };
-        // Add the modal property to the spotlight to mimic about:welcome
-        spotlightFake.content.modal = "tab";
-        // The recipe might have a backdrop, but if not, fall back to the default
-        spotlightFake.content.backdrop = feature.value.backdrop ||
-        "var(--mr-welcome-background-color) var(--mr-welcome-background-gradient)";
-        // Localize the recipe if necessary.
-        let localizedWelcome = _substituteLocalizations(
-          spotlightFake,
-          this._rawRecipe.localizations?.[
-            Object.keys(this._rawRecipe.localizations)[0]
-          ],
-        );
-
-        branchInfo.previewLink = getPreviewLink(localizedWelcome);
+        // Only create a preview object if there's something to preview
+        if (feature.value.hasOwnProperty("screens")) {
+          // featureValue will become the "content" object in a spotlight JSON
+          let spotlightFake = {
+            id: this._rawRecipe.id,
+            template: "spotlight",
+            targeting: true,
+            content: feature.value,
+          };
+          // Add the modal property to the spotlight to mimic about:welcome
+          spotlightFake.content.modal = "tab";
+          // The recipe might have a backdrop, but if not, fall back to the default
+          spotlightFake.content.backdrop = feature.value.backdrop ||
+          "var(--mr-welcome-background-color) var(--mr-welcome-background-gradient)";
+          // Localize the recipe if necessary.
+          let localizedWelcome = _substituteLocalizations(
+            spotlightFake,
+            this._rawRecipe.localizations?.[
+              Object.keys(this._rawRecipe.localizations)[0]
+            ],
+          );
+
+          branchInfo.previewLink = getPreviewLink(localizedWelcome);
+        }
         break;
 
       case 'feature_callout':
         // XXX should iterate over all screens
         branchInfo.id = feature.value.content.screens[0].id
+        // Localize the feature callout if necessary
+        console.log(this._rawRecipe);
+        if(this._rawRecipe.localizations) {
+          console.log(Object.keys(this._rawRecipe.localizations)[0]);
+        }
+        let localizedFeatureCallout = _substituteLocalizations(feature.value,
+this._rawRecipe.localizations?.[Object.keys(this._rawRecipe.localizations)[0]])
+        //console.log(JSON.stringify(localizedFeatureCallout, null, 2));
```

Comment:
Just to clean up the code, perhaps this comment can be removed:
```suggestion
```

Reply:
Oh yep, good catch!